http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package api
+
+import _root_.java.time.Instant
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server
+import akka.http.scaladsl.server.Directives._
+import cats.effect.{IO, Timer}
+import cats.implicits._
+import cats.mtl.ApplicativeAsk
+import org.broadinstitute.dsde.workbench.leonardo.http.api.LeoRoutes.validateRuntimeNameDirective
+import org.broadinstitute.dsde.workbench.leonardo.http.service.RuntimeService
+import org.broadinstitute.dsde.workbench.leonardo.http.service.RuntimeService.CreateRuntimeRequest
+import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.concurrent.ExecutionContext
+
+class RuntimeRoutes(runtimeService: RuntimeService[IO])(implicit timer: Timer[IO], ec: ExecutionContext) extends UserInfoDirectives {
+  def runtimeRoutes: server.Route = requireUserInfo { userInfo =>
+    implicit val traceId = ApplicativeAsk.const[IO, TraceId](TraceId(UUID.randomUUID()))
+    pathPrefix("google/v1/runtime") {
+      pathPrefix(googleProjectSegment / Segment) { (googleProject, runtimeNameString) =>
+        validateRuntimeNameDirective(runtimeNameString) { runtimeName =>
+          pathEndOrSingleSlash {
+            post {
```

Comment:
create runtime should be `PUT`

Reply:
I always thought `POST` is for creating a new entity, `PUT` is for updating a whole entity. Why do we want `PUT` for creating a runtime?

Reply:
Hm it does sound like it should be `POST` because it's not idempotent. TIL

https://restfulapi.net/rest-put-vs-post/
https://stackoverflow.com/questions/107390/whats-the-difference-between-a-post-and-a-put-http-request